File: src/main/scala/prc94/lavafurnace/LavaFurnaceBlock.scala
package prc94.lavafurnace

import javax.annotation.Nonnull
import net.minecraft.block.{Block, FurnaceBlock}
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.inventory.container.INamedContainerProvider
import net.minecraft.stats.Stats
import net.minecraft.util.math.BlockPos
import net.minecraft.world.{IBlockReader, World}


class LavaFurnaceBlock(builder: Block.Properties) extends FurnaceBlock(builder) {

  override def createNewTileEntity(worldIn: IBlockReader) = new LavaFurnaceTileEntity

  override protected def interactWith(worldIn: World, @Nonnull pos: BlockPos, @Nonnull player: PlayerEntity): Unit = {
    val tileentity = worldIn.getTileEntity(pos)
    if (tileentity.isInstanceOf[LavaFurnaceTileEntity]) {
      player.openContainer(tileentity.asInstanceOf[INamedContainerProvider])
      player.addStat(Stats.INTERACT_WITH_FURNACE)
    }
  }

}

File: src/main/kotlin/prc94/lavafurnacemod/lavafurnace/LavaFurnaceBlock.kt
package prc94.lavafurnacemod.lavafurnace

import net.minecraft.block.FurnaceBlock
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.stats.Stats.INTERACT_WITH_FURNACE
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockReader
import net.minecraft.world.World

class LavaFurnaceBlock(builder: Properties) : FurnaceBlock(builder) {

    override fun createNewTileEntity(worldIn: IBlockReader) = LavaFurnaceTileEntity()

    override fun interactWith(worldIn: World, pos: BlockPos, player: PlayerEntity) =
            worldIn.getTileEntity(pos).let {
                if (it is LavaFurnaceTileEntity) {
                    player.openContainer(it)
                    player.addStat(INTERACT_WITH_FURNACE)
                }
            }
}